### ISSUE TITLE
Result reproducibility

### ISSUE THREAD

**Issue by hmate9** [oliverguhr/transformer-time-series-prediction#25]
I have not been able to get the same same results as those presented here. Not even after 1000 epochs.

In the code the number if epochs is 10 and 100 for single and multi respectively. I was wondering if those were the lengths you trained them when posting the results?

**Comment by CrazyCyberbug** [comment]
Hi have you been able to find any solution to the problem? I found that both the variation of the model give a straight line parallel to the x-axis when working with the temperature dataset

**Comment by hmate9** [comment]
> Hi have you been able to find any solution to the problem? I found that both the variation of the model give a straight line parallel to the x-axis when working with the temperature dataset

No I never got around to it. I got the same result you did.

**Comment by micdic** [comment]
Same problem for me. Even if the model seems to work also tweaking a little bit the dummy dataset (i.e. changing amplitude or sinusoidal data trend), with the temperature dataset there is something wrong.... (straight line on the x-axis).
Any hints from the author?

**Comment by oliverguhr** [comment]
I worked on this years ago and never touched it again. Most likely, there are quirks in the code or architecture that lead to issues - after all, it's a proof of concept. I will take a look at this, but it will take some time. 

I would suggest trying models like [Autoformer](https://huggingface.co/blog/autoformer) or similar approaches.

**Comment by mdarm** [comment]
> I have not been able to get the same same results as those presented here. Not even after 1000 epochs.
> 
> In the code the number if epochs is 10 and 100 for single and multi respectively. I was wondering if those were the lengths you trained them when posting the results?

Initialize the learning rate value with 0.0005 instead. You can experiment with different values of course or with the learning_scheduler (gamma decay, steps etc.).
